content/components/icon.mdx

```diff
@@ -0,0 +1,216 @@
+---
+title: Icon
+react: https://primer.style/octicons/packages/react
+rails: https://primer.style/octicons/packages/ruby
+figma: https://www.figma.com/file/1ljgTFkT5NKNRfq5hw07JQ/Octicons?node-id=1%3A2&t=Lz5afoq1bbWrD9Yt-1
+description: Our icons are located in a separate library called Octicons, which is available in various implementations including React, Figma, Rails, and Styled System.
+---
+
+import {Box, Text} from '@primer/react'
+
+<img
+  width="960"
+  alt="An image of multiple icons placed in a grid"
+  src="https://user-images.githubusercontent.com/980622/210349372-9392f4b3-4902-4fd3-ab6f-fbd3587f208a.png"
+/>
+
+## Library
+
+At Primer, we utilize the [Octicons](/octicons/) icon library. This package is available in [Ruby](/octicons/packages/ruby), [Jekyll](/octicons/packages/jekyll), [JavaScript](/octicons/packages/javascript), [React](/octicons/packages/react) and [Styled System](/octicons/packages/styled-system).
+
+## Usage
+
+Icons are most appropriate for representing simple, recognizable concepts or actions. When choosing icons for your design system, consider the following guidelines:
+
+1. Where possible use icons to supplement text, rather than replacing it.
+2. Be sure that the meaning of icons is clear when used without accompanying text..
+3. Choose icons that are easily recognizable and accurately depict the intended concept or action.
+4. Maintain consistency in the use of icons across your product or brand to establish a cohesive visual language.
+5. Avoid using icons for complex or abstract concepts that may be confusing to some users.
+6. Adhere to predetermined icon sizes to ensure legibility and ease of recognition.
+7. Test the effectiveness of your icons with users to ensure that they are understood and useful in navigating the interface.
+
+## Color
+
+To ensure that icons are effective and legible in different color modes, it is recommended to use our [functional foreground colors](/foundations/color#foregrounds). This helps to maintain consistency and ensures that the icons convey the intended meaning to users.
+
+Many of our components are designed to automatically set the color of icons to a predetermined color. For example, flash banners are configured to set the icon to our [fg.danger](/foundations/color#foregrounds) functional color by default.
+
+<img
+  width="960"
+  alt="A flash banner showing a warning icon in orange with the text 'Your password was reset recently.'"
+  src="https://user-images.githubusercontent.com/980622/210349549-e7444c43-bb52-4a84-b00e-13962a22377a.png"
+/>
+
+In some cases, certain icons should always be paired with specific colors in order to effectively convey their intended meaning and provide a consistent user experience. Here are a few examples:
```

Comment:
This may be easier to read (or not, not 100% sure 😅)

```suggestion
Some icons should always be shown in a specific colors when used within a specific context. This helps to effectively convey their intended meaning and provide a consistent user experience. Here are a few examples:
```